packages/x-data-grid-premium/src/DataGridPremium/useDataGridPremiumProps.ts

```diff
@@ -35,14 +35,22 @@ export const DATA_GRID_PREMIUM_PROPS_DEFAULT_VALUES: DataGridPremiumPropsWithDef
 const defaultSlots = DATA_GRID_PREMIUM_DEFAULT_SLOTS_COMPONENTS;
 
 export const useDataGridPremiumProps = (inProps: DataGridPremiumProps) => {
-  const themedProps = useProps(
+  let themedProps = useProps(
     // eslint-disable-next-line material-ui/mui-name-matches-component-name
     useThemeProps({
       props: inProps,
       name: 'MuiDataGrid',
     }),
   );
 
+  const { pivotParams } = inProps;
+  if (pivotParams?.pivotMode) {
+    themedProps = {
+      ...themedProps,
+      ...pivotParams.props,
+    };
+  }
```

Comment:
The `unstable_useGridPivoting` hook is transforming the data and returns props overrides that we apply here.
The props are:
- `rows`
- `columns`
- `rowGroupingModel`
- `aggregationModel`
- `getAggregationPosition`
- `columnVisibilityModel`
- `columnGroupingModel`

These overrides seem like the most straightforward way to change the data grid configuration, but I'm open to feedback and ideas!

Reply:
Does that mean the pivot feature only be used in controlled mode?